compiler/rustc_trait_selection/src/solve/eval_ctxt/canonical.rs

```diff
@@ -382,6 +382,17 @@ impl<'tcx> TypeFolder<TyCtxt<'tcx>> for EagerResolver<'_, 'tcx> {
                     }
                 }
             }
+            ty::ConstKind::Infer(ty::InferConst::EffectVar(vid)) => {
+                // FIXME: we need to fold the ty too, I think.
```

Comment:
The effect variable always has type `bool` so we should be able to just use that here.

Reply:
Oh right, I can remove this fixme. It was just copy paste error.